### ISSUE TITLE
Changes needed in resteasy-reactive by RESTEasy

### ISSUE THREAD

**Issue by rsearls** [quarkusio/quarkus#16024]
## Description

In order for RESTEasy to use resteasy-reactive classes one structural change
is needed to the module. Class RuntimeDelegateImpl and file javax.ws.rs.ext.RuntimeDelegate
must be isolated in a module of its own.  This is because the JAX-RS API loads 
the first RuntimeDelegate it finds on the classpath.  To ensure RESTEasy's
implementation is used and not resteasy-reactive's the new module must be
explicitly excluded when referencing the resteasy-reactive archives during
the build process. 

Also there are a couple of minor changes need to couple of classes so that
RESTEasy can subclass them.

**Comment by geoand** [comment]
Thanks!

You mentioned I think in the email to the Dev list that you have a PR in mind.
Can you open it in draft state so we can see what this looks like in practice?

**Comment by michalszynkiewicz** [comment]
Why do we want to mix RESTEasy and RESTEasy Reactive? Or am I misundersting the intent?

**Comment by FroMage** [comment]
Yeah, I'm not really following what problem you're trying to solve, could you rephrase it please?

**Comment by rsearls** [comment]
> Thanks!
> 
> You mentioned I think in the email to the Dev list that you have a PR in mind.
> Can you open it in draft state so we can see what this looks like in practice?
And how does one do that?

**Comment by rsearls** [comment]
> Why do we want to mix RESTEasy and RESTEasy Reactive? Or am I misundersting the intent?
From doc,
https://docs.google.com/document/d/1VdCGPjig_aBM7eTDVXAhwVYjFqsZMZrEPdeyoz4doc4/edit

> The current RESTEasy team as well as the external community will be encouraged 
to contribute as well; the current RESTEasy team will also evaluate back and 
forward backports of fixes and features that are not completely tied to the 
backend architecture / design. **We’ll strive to avoid duplication of efforts 
to implement new spec versions, possibly isolating shared pieces in common subprojects.**

30+ classes were copied from RESTEasy into resteasy-reactive.
25 classes are identical except for the package name. 5+ of the classes
contain resteasy-reactive specific changes.

**Comment by michalszynkiewicz** [comment]
I don't have access to the doc, I requested it. If I'm asking already answered questions, please let me know :)

Will we have some common core project that would contain the 25 classes then? If we're going to pull out some common code, I think it would be best done this way so that we can release it independently of both RESTEasy and RESTEasy Reactive.

**Comment by rsearls** [comment]
> Yeah, I'm not really following what problem you're trying to solve, could you rephrase it please?

8 of the classes copied from RESTEasy extend from 

> RuntimeDelegateImpl extends javax.ws.rs.ext.RuntimeDelegate {
......
}

(e.g. NewCookieHeaderDelegate, CacheControlDelegate, DateDelegate, EntityTagDelegate,
LinkDelegate, LocaleDelegate, MediaTypeHeaderDelegate, CookieHeaderDelegate)

RESTEasy and resteasy-reactive have different implementations of RuntimeDelegateImpl.
They can't be shared.  Implementations of javax.ws.rs.ext.RuntimeDelegate are
declared as a service (e.g. META-INF/services/javax.ws.rs.ext.RuntimeDelegate)
The first service found on the classpath is use.  When including archive 
resteasy-reactive-common in RESTEasy, resteasy-reactive's RuntimeDelegateImpl 
is first in the classpath and is being used rather than RESTEasy's impl. This 
breaks RESTEasy.  RESTEasy needs a way to exclude  resteasy-reactive's
RuntimeDelegateImpl service.  The way to do that is to move resteasy-reactive's
RuntimeDelegateImpl out of module resteasy-reactive/common/runtime into one
of its own.  resteasy-reactive/common/runtime will include the module in its
build and RESTEasy will exclude it from its archive reference.  This should
allow resteasy-reactive to run with its version of RuntimeDelegate and RESTEasy 
to run with it's version and share the common *Delegate classes.

**Comment by rsearls** [comment]
> I don't have access to the doc, I requested it. If I'm asking already answered questions, please let me know :)
> 
> Will we have some common core project that would contain the 25 classes then? If we're going to pull out some common code, I think it would be best done this way so that we can release it independently of both RESTEasy and RESTEasy Reactive.

Quarkus still owns this project.  I think it is for them to say at this point.

**Comment by geoand** [comment]
We will soon move the core RESTEasy Reactive project out of the Quarkus repo, but the plan isn't to reuse code from RESTEasy, so I'll close this
